### DISCUSSION TITLE
cannot map zero-fill pages

### DISCUSSION THREAD

**Discussion by zsh151** [neams-th-coe/cardinal#856] [Q&A] [answered]
I'm trying to run the `MSFR` example in `Tutorials`. There was an error that I couldn't understand.
I have successfully run `SFR pincell`, so I don't think this should be an installation issue.

```
                __    ____  _____
   ____   ___   / /__ / __ \/ ___/
  / __ \ / _ \ / //_// /_/ /\__ \ 
 / / / //  __// ,<  / _, _/___/ / 
/_/ /_/ \___//_/|_|/_/ |_|/____/  v23.1.1 (48e408eb)

COPYRIGHT (c) 2019-2023 UCHICAGO ARGONNE, LLC

MPI tasks: 8

Initializing device 
general::filtering is deprecated and might be removed in the future!
general::filtermodes is deprecated and might be removed in the future!
general::filterweight is deprecated and might be removed in the future!
active occa mode: Serial

using NEKRS_HOME: /home/en/cardinal/install
using NEKRS_CACHE_DIR: /home/en/cardinal/tutorials/msfr/.cache
using OCCA_CACHE_DIR: /home/en/cardinal/tutorials/msfr/.cache/occa/

Error in set_usr_handles: /home/en/cardinal/tutorials/msfr/.cache/nek5000/libmsfr.so: cannot map zero-fill pages
Error in set_usr_handles: Error in set_usr_handles: /home/en/cardinal/tutorials/msfr/.cache/nek5000/libmsfr.so: cannot map zero-fill pages
skip building nekInterface (SIZE requires no update)
loading nek ... Error in set_usr_handles: /home/en/cardinal/tutorials/msfr/.cache/nek5000/libmsfr.so: cannot map zero-fill pages
/home/en/cardinal/tutorials/msfr/.cache/nek5000/libmsfr.so: cannot map zero-fill pages
Error in set_usr_handles: /home/en/cardinal/tutorials/msfr/.cache/nek5000/libmsfr.so: cannot map zero-fill pages
Error in set_usr_handles: /home/en/cardinal/tutorials/msfr/.cache/nek5000/libmsfr.so: cannot map zero-fill pages
Error in set_usr_handles: Error in set_usr_handles: /home/en/cardinal/tutorials/msfr/.cache/nek5000/libmsfr.so: cannot map zero-fill pages
/home/en/cardinal/tutorials/msfr/.cache/nek5000/libmsfr.so: cannot map zero-fill pages

```

Here is the output file.
[output.txt](https://github.com/neams-th-coe/cardinal/files/14481603/output.txt)

PS I have downloaded mesh and restart files from Box.
![图片2](https://github.com/neams-th-coe/cardinal/assets/110072634/dc20651f-f87c-49d5-bd35-d6ac319bbf4a)

**Comment by aprilnovak** [comment, marked as answer]
As mentioned on the tutorial page, "This tutorial requires High-Performance Computing (HPC) for running the NekRS model. You will be able to run the OpenMC model without HPC resources." The NekRS model is a full-core large eddy simulation, and will run out of memory if your system has 8 cores. I ran this model on about 30 nodes of Summit, or 180 GPUs, for reference. 

Do you have access to a cluster? If so, you may want to check out some of our pre-existing job scripts (https://cardinal.cels.anl.gov/hpc.html) for Cardinal runs for some other systems, and try adapting them.

**Reply by zsh151** [reply]
I see, thanks.
So does this `cannot map zero-fill pages` means NekRS run out of memory?

**Reply by aprilnovak** [reply]
When NekRS runs out of memory, the program generally just aborts in whatever routine it happened to be in -- so the message printed when running out of memory can take a lot of different forms. But I am 99% sure that it is failing due to lack of memory, since that case is quite large (and you were able to run the sfr_pincell one okay).

**Reply by zsh151** [reply]
Okay, thanks a lot
